pkg/cmd/issue/transfer/transfer.go

```diff
@@ -0,0 +1,168 @@
+package transfer
+
+import (
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/cmd/issue/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/cli/cli/pkg/text"
+	"github.com/spf13/cobra"
+)
+
+type TransferOptions struct {
+	HttpClient func() (*http.Client, error)
+	Config     func() (config.Config, error)
+	IO         *iostreams.IOStreams
+	BaseRepo   func() (ghrepo.Interface, error)
+
+	IssueSelector    string
+	DestRepoSelector string
+}
+
+func NewCmdTransfer(f *cmdutil.Factory, runF func(*TransferOptions) error) *cobra.Command {
+	opts := TransferOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+		Config:     f.Config,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "transfer {<number> | <url>} <destination-repo>",
+		Short: "Transfer issue",
+		Args:  cobra.MinimumNArgs(1),
+		Example: heredoc.Doc(`
+					$ gh issue 3252 
+					$ gh issue 3252 cli/cli
+					$ gh issue 3252 --repo cli/cli golang/go
+					$ gh issue 3252 github.com/golang/go
+				`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+			if len(args) > 0 {
+				opts.IssueSelector = args[0]
+			}
+
+			if len(args) > 1 {
+				opts.DestRepoSelector = args[1]
+			}
+
+			if runF != nil {
+				return runF(&opts)
+			}
+
+			return transferRun(&opts)
+		},
+	}
+
+	return cmd
+}
+
+func transferRun(opts *TransferOptions) error {
+	client, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	apiClient := api.NewClientFromHTTP(client)
+
+	issue, baseRepo, err := shared.IssueFromArg(apiClient, opts.BaseRepo, opts.IssueSelector)
+	if err != nil {
+		return err
+	}
+
+	srcRepo, err := api.GitHubRepo(apiClient, baseRepo)
+	if err != nil {
+		return err
+	}
+
+	if !srcRepo.ViewerCanPush() {
+		return fmt.Errorf("You don't have WRITE permission on %s/%s", srcRepo.RepoOwner(), srcRepo.RepoName())
```

Comment:
Let's avoid doing this check and let the API inform us of whether we are able to transfer the issue or not. The rules of when you are able to transfer the issue might change on GitHub and we don't want to be keeping these rules on the client in sync.

By deleting this, we also don't have to look up `srcRepo` at all.